pkg/storage/segments/store.go

```diff
@@ -271,6 +273,128 @@ func (s *segmentStore) Delete(ctx context.Context, path paths.Path) (err error)
 	return s.pdb.Delete(ctx, path)
 }
 
+// Repair retrieves an at-risk segment and repairs and stores lost pieces on new nodes
+func (s *segmentStore) Repair(ctx context.Context, path paths.Path, lostPieces []int) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	//Read the segment's pointer's info from the PointerDB
+	pr, err := s.pdb.Get(ctx, path)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	if pr.GetType() != pb.Pointer_REMOTE {
+		return Error.New("Cannot repair inline segment")
+	}
+
+	seg := pr.GetRemote()
+	pid := client.PieceID(seg.GetPieceId())
+
+	// Get the list of remote pieces from the pointer
+	originalNodes, err := s.lookupNodes(ctx, seg)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	// get the nodes list that needs to be excluded
+	var excludeNodeIDs []dht.NodeID
+	for _, v := range originalNodes {
```

Comment:
We should be able to combine all of the for loops

Reply:
combined all the for loops - optimized